redbeams-api/src/main/java/com/sequenceiq/redbeams/validation/DatabaseVendorAndServiceValidator.java

```diff
@@ -0,0 +1,62 @@
+package com.sequenceiq.redbeams.validation;
+
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+import com.sequenceiq.cloudbreak.api.endpoint.v4.common.DatabaseVendor;
+import com.sequenceiq.cloudbreak.validation.ValidatorUtil;
+import com.sequenceiq.cloudbreak.validation.externaldatabase.SupportedDatabaseEntry;
+import com.sequenceiq.cloudbreak.validation.externaldatabase.SupportedDatabaseProvider;
+import com.sequenceiq.cloudbreak.validation.externaldatabase.SupportedExternalDatabaseServiceEntry;
+import com.sequenceiq.redbeams.api.endpoint.v4.database.request.DatabaseV4Request;
+import com.sequenceiq.redbeams.api.util.DatabaseVendorUtil;
+
+public class DatabaseVendorAndServiceValidator implements ConstraintValidator<ValidDatabaseVendorAndService, DatabaseV4Request> {
+
+    private DatabaseVendorUtil databaseVendorUtil = new DatabaseVendorUtil();
+
+    @Override
+    public boolean isValid(DatabaseV4Request request, ConstraintValidatorContext context) {
+        Optional<DatabaseVendor> vendorByJdbcUrl = databaseVendorUtil.getVendorByJdbcUrl(request);
+
+        Optional<SupportedExternalDatabaseServiceEntry> serviceEntry = SupportedDatabaseProvider.supportedExternalDatabases()
+                .stream()
+                .filter(item -> item.getName().equalsIgnoreCase(request.getType())
+                        || item.getDisplayName().equalsIgnoreCase(request.getType()))
+                .findFirst();
+        if (!serviceEntry.isPresent()) {
+            serviceEntry = SupportedDatabaseProvider.getOthers();
+        }
+
+        if (serviceEntry.isPresent() && vendorByJdbcUrl.isPresent()) {
+            Set<SupportedDatabaseEntry> allSupportedDatabases = serviceEntry.get().getDatabases();
+            boolean supported = allSupportedDatabases.stream()
+                    .anyMatch(item -> item.getDatabaseName().equals(vendorByJdbcUrl.get().name()));
+            if (supported) {
+                return true;
+            } else {
```

Comment:
I think the else clause is not needed as the code returns from the other branch.

Reply:
correct